### PR TITLE
feat(popover): allow to set popover trigger

### LINKED ISSUE #3116: Allow to trigger Popover on mouseenter
Hi,

In Kiali we have a regression with PF4, as it seems popovers can't be triggered on mouse-enter anymore (or I didn't find how).
Is it done on purpose? It would be nice to get it back.

Before you ask, I've also considered using `Tooltip` instead of `Popover` but it doesn't really look well because it's not short content (as supposed to be in tooltips) and also tooltip text is centered which is not what I want.

Related PR in Kiali: https://github.com/kiali/kiali-ui/pull/1475

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

closes #3116 

I added the `triggers` prop to modify the trigger option passed to `TippsyJS`.

**Please notice** that if you set `isVisible` this props is irrelevant since you are the one who manages when to display the popover and thus you will have to implement those triggers or catch those events and mutate the state that determines whether to display or hide the popover.

//cc @jotak

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Popover/Popover.tsx [hunk 1]
```diff
@@ -69,6 +78,8 @@ export interface PopoverProps {
    * Instead, the consumer is responsible for closing the popover themselves by adding a callback listener for the shouldClose prop.
    */
   isVisible?: boolean;
+  /** The events which cause the popover to show. **/
+  triggers?: Triggers[];
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Popover/Popover.tsx]: Can you please note that is isVisible is set, this prop is ignored.

File: packages/patternfly-4/react-core/src/components/Popover/Popover.test.tsx [hunk 2]
```diff
@@ -62,3 +62,20 @@ test('popover passes along values to tippy.js', () => {
   );
   expect(view).toMatchSnapshot();
 });
+
+test('popover with custom triggers', () => {
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Popover/Popover.test.tsx]: this is great.  Can you also add a demo to the demo app and an integration test for this.

File: packages/patternfly-4/react-core/src/components/Popover/Popover.tsx (hunk 1 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Popover/Popover.tsx]: Good idea!

File: packages/patternfly-4/react-core/src/components/Popover/Popover.test.tsx (hunk 2 above)
Comment (boaz0) [reply packages/patternfly-4/react-core/src/components/Popover/Popover.test.tsx]: Yeah sure! I added a test into its spec to make sure that setting a focus trigger will display the popover. :man_technologist:

**Comment by jotak** [comment]
Sorry I was on PTO . Thanks @boaz0  for taking this. I'll check asap.

**Comment by jotak** [comment]
It looks good to me!

**Comment by jotak** [comment]
@mcarrano it happens that we actually moved to `Tooltip` just yesterday but for a different reason, more as a workaround to another issue we had with Popover: https://github.com/kiali/kiali-ui/pull/1521

**Comment by jgiardino** [comment]
> I do understand that our tooltip is very constrained and may not be giving you the flexibility you need for formatting the content. If we are to make a change, I would rather update the tooltip component to accept formatted content. @mcoker can you take a look at this issue kiali/kiali-ui#1475 What would it take to add that content to a tooltip? If that seems like a reasonable approach, I can open an issue in core to create that variant.

Regarding this comment from @mcarrano, I posted a [comment in the original issue](https://github.com/patternfly/patternfly-react/issues/3116#issuecomment-550487833) with an example of what the Tooltip currently supports.

**Comment by mcoker** [comment]
@jgiardino @mcarrano technically you can put anything you want in the tooltip content as far as content goes, though as @jgiardino discovered, there may be font color issues as we set that explicitly. Instead, we could assign dark theme variable substitutions inside of the tooltip content, which would turn anything that uses our dark text color white inside of the tooltip content area. That change would happen in core and would go into react with no updates other than pulling in the core CSS changes.

And to mimic that behavior currently, you could use the class `.pf-t-dark` on an element that wraps the toolbar content, and that will do the same variable substitutions, resulting in anything that uses our default text color changing white in the tooltip content area.

**Comment by mcarrano** [comment]
@boaz0 @jotak Given all of this, can we close this PR? If we want to follow up on improvements to the Tooltip component, a new issue should be opened for that.

**Comment by boaz0** [comment]
Sure, let me open a follow up ticket for that.

**Pull request closed without merging**